receiver/datadogreceiver/receiver_test.go

```diff
@@ -85,3 +87,13 @@ func TestDatadogServer(t *testing.T) {
 		})
 	}
 }
+
+// Throughout the repo, many tests are invoked with go test ... -parallel ...
+// This means that in addition to explicitly parallelizable tests (t.Parallel()), most test cases can be executed
+// in parallel. Components that statically try to bind a port (e.g., from an integration test) can therefore fail
+// and cause flaky builds. This function wraps Component.start with a simple ≤5s retry schedule.
+func startComponentWithRetries(ctx context.Context, component component.Component, host component.Host) error {
+	return backoff.Retry(func() error {
+		return component.Start(ctx, host)
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Second), 5))
```

Comment:
Maintainers: feel free to fiddle with the parameters here (e.g., feel free to edit on-PR yourself). I just wanted to point out the flake and propose a fix. I wouldn't want to use too many retries to slow down overall build times and mask an issue that should be fixed more structurally. However, I think that it makes sense for the integration test to use the expected standard port instead of a randomly generated (guaranteed-not-in-use) port.

Reply:
Or, alternatively, we can just mash the two test cases (functions) in this suite (file) into one which will guarantee serial execution of their steps regardless of `-parallel N` invocations of `gotest` in CI.